### DISCUSSION TITLE
Stopping abci.socketClient for error: EOF

### DISCUSSION THREAD

**Discussion by Nkorchi** [tendermint/tendermint#9638] [Q&A]
Hi,

I am running BigchainDB which uses MongoDB and Tendermint for consensus. BigchainDB supports Tendermint version not older than v0.31.5. I have created 4 VMs in Digital Ocean Cloud, Ubuntu 18.04, and made all the configurations that are described [HERE](http://docs.bigchaindb.com/en/latest/installation/network-setup/network-setup.html) . The application starts just fine and Tendermint nodes also connect to each other and start validating blocks as the application receives the incoming transactions. But at some point, usually halfway, it raises this error and stops the consensus and node respectively. 

I ran this set of experiments on 4 other nodes with the same OS on Digital Ocean CLoud but only difference was those machines had less CPUs and Memory. I upgraded to CPU optimized machines to get higher computing power but this error started arising. 

Can anyone help to tackle this issue? 

Thank you! 

`E[2022-10-27|00:21:35.922] Stopping abci.socketClient for error: EOF    module=abci-client connection=consensus`
`E[2022-10-27|00:21:35.923] Error in proxyAppConn.EndBlock               module=state err=EOF`
`E[2022-10-27|00:21:35.923] Error on ApplyBlock. Did the application crash? Please restart tendermint module=consensus err=EOF`

`I[2022-10-27|00:21:35.923] captured terminated, exiting...              module=main `
`I[2022-10-27|00:21:35.923] Stopping Node                                module=main impl=Node`
`I[2022-10-27|00:21:35.923] Stopping Node                                module=main `

`E[2022-10-27|00:21:35.923] Connection failed @ recvRoutine (reading byte) module=p2p 
peer=a7aad8a73dc64e984558bf5d491bf5f8742cc1a6@68.183.123.228:52974 conn=MConn{68.183.123.228:52974} err="read tcp 157.230.238.183:26656->68.183.123.228:52974: use of closed network connection"`

`E[2022-10-27|00:21:35.923] Stopping peer for error                      module=p2p peer="Peer{MConn{68.183.123.228:52974} a7aad8a73dc64e984558bf5d491bf5f8742cc1a6 in}" err="read tcp 157.230.238.183:26656->68.183.123.228:52974: use of closed network connection"`

`E[2022-10-27|00:21:35.923] Connection failed @ recvRoutine (reading byte) module=p2p peer=8ea5f3ba674f3ddf3315a582a92c3766e485badf@68.183.107.239:47520 conn=MConn{68.183.107.239:47520} err="read tcp 157.230.238.183:26656->68.183.107.239:47520: use of closed network connection"`

`E[2022-10-27|00:21:35.923] Stopping peer for error                      module=p2p peer="Peer{MConn{68.183.107.239:47520} 8ea5f3ba674f3ddf3315a582a92c3766e485badf in}" err="read tcp 157.230.238.183:26656->68.183.107.239:47520: use of closed network connection"`

`E[2022-10-27|00:21:35.923] Connection failed @ recvRoutine (reading byte) module=p2p peer=d5a1894e4a162248e82c11efb6d0c046d2881468@157.230.225.152:44792 conn=MConn{157.230.225.152:44792} err=EOF`

`E[2022-10-27|00:21:35.923] Stopping peer for error                      module=p2p peer="Peer{MConn{157.230.225.152:44792} d5a1894e4a162248e82c11efb6d0c046d2881468 in}" err=EOF`

`E[2022-10-27|00:21:35.924] Stopped accept routine, as transport is closed module=p2p numPeers=0`

`I[2022-10-27|00:21:35.924] Closing rpc listener                         module=main listener="&{Listener:0xc0004a21e0 sem:0xc000146b40 closeOnce:{m:{state:0 sema:0} done:0} done:0xc000146ba0}`

**Comment by thanethomson** [comment]
The oldest version of Tendermint we currently support is the v0.34 series. What version of Tendermint are you running?

**Reply by Nkorchi** [reply]
I am running v0.31.5.

**Reply by thanethomson** [reply]
Is there any way of upgrading to v0.34? Otherwise we won't really be able to help.
